src/core/Address.ts

```diff
@@ -79,29 +77,4 @@ export abstract class Address {
     public toString(): string {
         return this.encoded;
     }
-
-    /**
-     * Determines the validity of an raw address string.
-     *
-     * @param encodedAddress - The raw address string. Expected format VATNE7Q5BITMUTRRN6IB4I7FLSDRDWZA35C4KNQ
-     * @returns true if the raw address string is valid, false otherwise.
-     */
-    public static isValid(encodedAddress: string): boolean {
-        const isSymbol = encodedAddress.length === 39;
-        if (isSymbol) {
-            if (!['A', 'I', 'Q', 'Y'].includes(encodedAddress.slice(-1).toUpperCase())) {
-                return false;
-            }
-            try {
-                const address = SymbolAddress.createFromString(encodedAddress);
-                const hasher = sha3_256.create();
-                const hash = hasher.update(address.rawAddress.addressWithoutChecksum).arrayBuffer();
-                return arrayDeepEqual(new Uint8Array(hash).subarray(0, 3), address.rawAddress.checksum);
-            } catch {
-                return false;
-            }
-        }
-        // TODO: Implement Nis1 Logic here
```

Comment:
The hash and checksum validate are pretty much the same except for the checksum byte length, why couldn't validate the address from the abstract class?

Reply:
yeah, but other than checksum bytes, the `address length`, `hash algorithm` is different as well.
I see nothing much code in common, so I move it to Nis1/Symbol class.

Reply:
probably can extract the varify code to private methods, passing in the hasher, hasher get decided by the length of the address (symbol/nis1) so that we can call Address.verify rather than calling the child classes

Reply:
Hi there, my 2 cents. 

What I don't like about having a top-level `Address.isValid(encodedAddress:string): boolean` is that the user may think a nis1 address is valid when the user is actually sending a symbol address or the other way around. The abstract doesn't work because the method is static, there is no object instance.

What I would do is having one isValid method per Subclass and remove the Ifs:

```
SymbolAddress.isValid(encodedAddress:string): boolean
Nis1Address.isValid(encodedAddress:string): boolean
```